geoutils/raster/raster.py

```diff
@@ -354,23 +355,78 @@ def _get_reproject_params(
     return dst_transform, dst_size
 
 
-def _check_cast_array_raster(
-    input1: RasterType | NDArrayNum, input2: RasterType | NDArrayNum, operation_name: str
-) -> None:
+def _cast_pixel_interpretation(
+    area_or_point1: Literal["Area", "Point"] | None, area_or_point2: Literal["Area", "Point"] | None
+) -> Literal["Area", "Point"] | None:
     """
-    Check the casting between an array and a raster, or raise an (helpful) error message.
+    Cast two pixel interpretations and warn if not castable.
 
-    :param input1: Raster or array.
-    :param input2: Raster or array.
+    Casts to:
+     - "Area" if both are "Area",
+     - "Point" if both are "Point",
+     - None if any of the interpretation is None, or
+     - None if one is "Area" and the other "Point" (and raises a warning).
+    """
+
+    # If one is None, cast to None
+    if area_or_point1 is None or area_or_point2 is None:
+        area_or_point_out = None
+    # If both are equal and not None
+    elif area_or_point1 == area_or_point2:
+        area_or_point_out = area_or_point1
+    else:
+        area_or_point_out = None
+        msg = (
+            'One raster has a pixel interpretation "Area" and the other "Point". To silence this warning, '
+            "either correct the pixel interpretation of one raster (see ), or deactivate "
```

Comment:
is there a missing text in the brackets, after "see ..."?

Reply:
Will add a link to the documentation here once done. Opened #523